libs/expo/betterangels/src/lib/screens/AddEditClient/RelevantContacts/Contact/index.tsx

```diff
@@ -0,0 +1,147 @@
+import { Colors, Radiuses, Spacings } from '@monorepo/expo/shared/static';
+import {
+  Input,
+  Select,
+  TextBold,
+  TextButton,
+} from '@monorepo/expo/shared/ui-components';
+import { useFormContext } from 'react-hook-form';
+import { View } from 'react-native';
+import {
+  CreateClientProfileInput,
+  RelationshipTypeEnum,
+  UpdateClientProfileInput,
+} from '../../../../apollo';
+import { enumDisplayRelevant } from '../../../../static/enumDisplayMapping';
+
+interface IContactProps {
+  index: number;
+  remove: (index: number) => void;
+}
+
+export default function Contact(props: IContactProps) {
+  const { index, remove } = props;
+  const { control, setValue, watch } = useFormContext<
+    UpdateClientProfileInput | CreateClientProfileInput
+  >();
+
+  const contacts = watch('contacts');
+  const relationship = watch(
+    `contacts[${index}].relationshipToClient` as `contacts.${number}.relationshipToClient`
+  );
+
+  const handleRemove = () => {
+    remove(index);
+    if (contacts?.length === 1) {
+      setValue('contacts', []);
+    }
```

Comment:
Can you explain this to me please? I see that it works but I'm not sure I understand how. If you have one contact, `contacts.length === 1`, when you `remove(index)` doesn't `contacts.length === 0`? I'm probably misunderstanding what operation is being performed on what

Reply:
don't need this function. previously emptying the state was cousing error. but now it works